### PR TITLE
Fix random behavior of CylinderStructured

### LINKED ISSUE #439: Control the seed point of CylinderStructured
Hi Bane,

thanks for this very useful method to quickly build a cylindrical array.

After getting more serious with it, I found out that the seed point can be randomly located. If we extract the cells and unravel to get numpy arrays, the location of the first array around the cylinder cannot be controlled. It would be very useful to specify a second direction to be able to control where the cell starts.
We can identify the first cell with proper opacity setting and because the points are repeated at the beginning and end of the grid.
image

### PR DESCRIPTION
**Pull request by banesullivan**
Resolve #439 

This should fix the random location of the seed point by not arbitrarily rotating the mesh to the given direction vector.

Someone may need to check my vector math

### CODE REVIEW

File: pyvista/utilities/geometric_objects.py [hunk 1]
```diff
@@ -161,14 +161,15 @@ def CylinderStructured(radius=0.5, height=1.0,
...
     # Orient properly in user direction
-    direction = np.array(direction)
-    direction = direction / np.linalg.norm(direction)
-    vx = np.random.randn(3)
-    vx -= vx.dot(direction) * direction
-    vx /= np.linalg.norm(vx)
-    vy = np.cross(direction, vx)
-    rmtx = np.array([vx, vy, direction])
-    grid.points = grid.points.dot(rmtx)
+    vx = np.array([0., 0., 1.])
```
Comment (banesullivan) [inline pyvista/utilities/geometric_objects.py]: This is the direction in which I created the structured grid to begin with, so my logical here is that if the direction is anything but this, then let's rotate from this vector to the given direction

File: pyvista/utilities/geometric_objects.py [hunk 2]
```diff
@@ -161,14 +161,15 @@ def CylinderStructured(radius=0.5, height=1.0,
...
+    if not np.allclose(vx, direction):
+        direction = np.array(direction)
```
Comment (prisae) [inline pyvista/utilities/geometric_objects.py]: What about `direction = np.array(direction, copy=False)` or  `direction = np.asarray(direction)` (the are basically the same).

Also, as you use `direction` anyway in the `if`, I would put the casting just one line before `if not np.allclose`.

Comment (prisae) [reply pyvista/utilities/geometric_objects.py]: The `asarray` or `copy=False` avoids making a new copy if it is already an array.

File: pyvista/utilities/geometric_objects.py [hunk 3]
```diff
@@ -161,14 +161,15 @@ def CylinderStructured(radius=0.5, height=1.0,
...
+        direction = direction / np.linalg.norm(direction)
```
Comment (imsodin) [inline pyvista/utilities/geometric_objects.py]:
```suggestion
        direction /= np.linalg.norm(direction)
```

File: pyvista/utilities/geometric_objects.py (hunk 2 above)
Comment (imsodin) [reply pyvista/utilities/geometric_objects.py]: Is it even necessary at all to explicitly convert direction? Numpy methods can take "array-like" objects and return ndarrays.

Comment (prisae) [reply pyvista/utilities/geometric_objects.py]: You are probably right. Might be one of those 'better safe than sorry'.

Comment (prisae) [reply pyvista/utilities/geometric_objects.py]: Looking at the entire function makes me think that you are right, casting is not required at all as it is done by numpy itself (problems can happen without casting if you try to access numpy-attributes like `dtype`, `max`).

Comment (banesullivan) [reply pyvista/utilities/geometric_objects.py]: @prisae, you point on 'better safe than sorry' is exactly why I did that - which was mostly just from habit. It shouldn't be needed for this particular function

```suggestion
```

File: pyvista/utilities/geometric_objects.py (hunk 3 above)
Comment (banesullivan) [reply pyvista/utilities/geometric_objects.py]: Ah thanks!

**Comment by banesullivan** [comment]
Thank you both @prisae and @imsodin for the quick reviews! 

@prisae, you've given this an approval, but you're not a member, so it doesn't register 😞

**Pull request merged**